src/documents/views.py

```diff
@@ -1203,7 +1205,7 @@ def get(self, request, *args, **kwargs):
                         s,
                         request.query_params,
                         10,
-                        request.user,
+                        filter_queryset=all_docs,
                     )._get_query()
                     results = s.search(q, limit=OBJECT_LIMIT)
                     docs = docs | all_docs.filter(id__in=[r["id"] for r in results])
```

Comment:
This is should keep the same behavior (actually `filter_queryset` is never used because `__getitem__` is never called), but IMO the existing implementation has some issues.

Let's say there are some documents not visible to the user, then `results` might get filled by those documents and the returned list might have fewer than 3 documents even though there are better candidates visible to the user (e.g., see the [updated test case in this commit](https://github.com/yichi-yang/paperless-ngx/commit/9e24f5c9aef5571d91f1458e09a92fa32cd0ffe3)).

If we are OK with changing this behavior (for the better), we can also include [this fix](https://github.com/yichi-yang/paperless-ngx/commit/9e24f5c9aef5571d91f1458e09a92fa32cd0ffe3) in this PR.

Reply:
Sure